src/main/java/ev3dev/sensors/Battery.java

```diff
@@ -67,38 +67,37 @@ private Battery() {
         currentRereader = new DataChannelRereader(batteryPath + "/" + batteryEv3 + "/" + current);
     }
 
-    public int getVoltageMicroVolts() {
+    public int getVoltageMicroVolt() {
         return Integer.parseInt(voltageRereader.readString());
     }
 
+    /**
+     * @return voltage of the battery in millivolts.
+     */
     @Override
     public int getVoltageMilliVolt() {
-        return getVoltageMicroVolts() / 1000;
+        return getVoltageMicroVolt() / 1000;
     }
 
     /**
-     * Returns voltage of the battery in microvolts.
-     *
-     * @return voltage
+     * @return voltage of the battery in microvolts.
      */
     public float getVoltage() {
-        return getVoltageMicroVolts() / 1000000f;
+        return getVoltageMicroVolt() / 1000000f;
     }
 
     //TODO Review output
     //TODO Review units
 
     /**
-     * Returns the current of the battery in amps.
-     *
-     * @return current
+     * @return current from the battery in amps, or Float.NaN if run on something other than EV3BRICK
      */
     public float getBatteryCurrent() {
         if (CURRENT_PLATFORM.equals(EV3DevPlatform.EV3BRICK)) {
             return Float.parseFloat(currentRereader.readString());
```

Comment:
This is unrelated to DataChannelRereader/DataChannelRewriter and as such it should be addressed in a different PR, but this method likely contains a bug. I'm wondering if this doesn't apply universally: if you're using `readFloat()` for a sysfs attribute, you're likely doing something wrong. Kernel doesn't usually work with floats, it works with scaled integers.

The problem is that the value will be read in microamps instead of amps. As per https://github.com/torvalds/linux/blob/master/Documentation/power/power_supply_class.rst:

> Quoting include/linux/power_supply.h:
>> All voltages, currents, charges, energies, time and temperatures in µV, µA, µAh, µWh, seconds and tenths of degree Celsius unless otherwise stated. It's driver's job to convert its raw values to units in which this class operates.

Reply:
Is that going to be a problem for sensor's GenericMode? It's filling arrays of floats.

The gyroscope only returns ints. I think the color sensor's API says it gives values between 0 and 1, but I haven't plugged ours in to verify that yet.

Reply:
GenericMode is luckily aware of the scaling, so while it reads floats from sysfs, it multiplies them by a scaling factor to get results in proper SI units. It should be possible to put read-integer there without breaking anything.

Reply:
There's an undercurrent of intent in the code to replace GenericMode with more specific modes. Maybe that's some future work to take up.

Floating point is pretty easy to start with, but turns out to be remarkably hard to explain to kids why == is a bad choice. Rational numbers in contrast are easy. With really small units - like millimeter, millisecond, or microvolt - it is really easy to explain why == is a bad choice. The real twist with GenericMode was "If I only want the number of degrees why do I have to pass in a float array of one float?"